src/main/java/com/google/graphgeckos/dashboard/components/BuildBotData.java

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.components;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.Timestamp;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.data.annotation.Transient;
+
+/**
+ * Contains the information retrieved from a single build bot. It is used as a member of BuildInfo...
+ * of {@link BuildInfo}
+ */
+@Entity(name = "builder")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BuildBotData {
+
+  @Transient
+  private String commitHash;
+
+  /* The timestamp of the build. */
+  private Timestamp timestamp;
+
+  /* Name of the buildbot. */
+  private final String name;
+
+  /**
+   * The logs of each compilation stage, stored as described by {@link Log}.
+   */
+  private final List<Log> logs = new ArrayList<>();
+
+  /**
+   * Builder compilation status, as described by {@link BuilderStatus}.
+   */
+  private BuilderStatus status;
+
+  /**
+   * Extracts nested commitHash("revision"), branch("branch") and timestamp("when")
+   * fields from the json.
+   * @param sourceStamp Representation of the json component, where the commitHash field is located.
+   */
+  @JsonProperty("sourceStamp")
+  public void unpackSourceStamp(Map<String, Object> sourceStamp) {
+    commitHash = sourceStamp.get("revision").toString();
+    timestamp = Timestamp.of(new Date(sourceStamp.get("when").toString()));
+  }
+
+  /**
+   * Defines builder status based on a phrase provided in parsed json.
+   * If something failed, then the buidbot is considered as failed {@code FAILED}.
+   * If something lost, then the buildbot is considered lost {@code LOST}.
```

Comment:
If something _is_ lost

Reply:
Also, I'd not say the buildbot is lost/failed/passed. I'd say the revision failed on the buildbot